flight/Modules/UAVOFrSKYSensorHubBridge/UAVOFrSKYSensorHubBridge.c

```diff
@@ -400,6 +400,9 @@ static void uavoFrSKYSensorHubBridgeTask(void *parameters)
 			// As long as there is no voltage for each cell
 			// all cells will have the same voltage.
 			// Receiver will know number of cells.
+			if(batState.DetectedCellCount)
+				batSettings.NbCells = batState.DetectedCellCount;
```

Comment:
Even if it takes a little more RAM I think it would be good to assign to a new variable rather than the settings struct, just for clarity.